### ISSUE TITLE
Allow to pin/bookmark items

### ISSUE THREAD

**Issue by p0deje** [p0deje/Maccy#26]
There were few requests to support pins/bookmarks of the history items. Here is one of them:

> I reply to all of our Yelp! Business requests.  They typically are inquiring about pricing and I have two formatted replies, which I select based on their inquiry.  I then of course edit it to make it more personal to their needs, but it’s handy to have most of the text added on a fly.
> 
> While I’m no developer I can offer some random suggestions.
> 
> +Maccy continues it’s clean UI.  I wouldn’t want any extra feature to hamper this.
> +“Pinned” results will always appear at the top when opening Maccy.
>   ++ This can either be with a pinned icon, star, etc.
>   ++ This also allows the top pinned results to always be the same command shortcut (unless you’d like the user to be able to add custom)
> + How you’d pin/favorite would be up to you! We both know there’s various ways, either through an addition menu, a submenu, etc.
>   ++ I think pins would be the most simple, but folders would apply organization.  The folder option may be too far in another direction though, as Maccy is great to search though.
> +If no pins, maybe a ’tag’ system? Instead of searching for words appropriate for my desired query, I can search the ’tag’.
>   ++ I want to find “work”.  I just type work, and it shows the results I tagged appropriately.

This issue is to track this feature and prototype it as there are multiple ways to approach this.

**Comment by gvieira** [comment]
I've been using [Alfred's Clipboard History](https://www.alfredapp.com/help/features/clipboard/) for ages, but I'm moving to Maccy now. Still, I would like to keep this link here as an idea on how they handle `Snippets`.

**Comment by LionelSelie** [comment]
Hi, I would love to have pinned items as well. iPaste currently does this, see this screenshot:

![7F20698B-0086-4393-80A9-EFF063C564C0](https://user-images.githubusercontent.com/53119148/63851539-10c77180-c99f-11e9-91ee-62d409d5ffa6.png)

The (1) is for normal clips, (2) are pinned items and (3) is a folder.

Also, it'd be awesome to have a system in place where you could rename the pinned item (also possible in iPaste). This could be interesting for calling a long piece of text a memorable name, e.g. snippet.

Basically introducing these options would make this a super powerhouse snippet manager, which is lightweight, yet flexible. I also do prefer such a system, compared to Alfred.

Thanks for considering it!

**Comment by p0deje** [comment]
I have a basic implementation of pins, which for now looks like this:

<img width="346" alt="Screen Shot 2020-02-23 at 09 39 30" src="https://user-images.githubusercontent.com/665846/75114013-96f80400-5620-11ea-83db-8a6fbc348171.png">

Basically you select any item you want and then click `Alt+p` to pin or unpin the item. Once pinned, it's assigned a random keyboard shortcut and moves to the top of the list.

Few questions however arise:

1. Should the pinned be on top or bottom?
2. If the pinned are on top and user opens Maccy, should the first pinned item be selected or the first unpinned?

**Comment by FernandoMiguel** [comment]
Pinned at the bottom, by offsetting very long list items, ie, it will be visible and crop the rest of the non pinned items

**Comment by p0deje** [comment]
> Pinned at the bottom, by offsetting very long list items, ie, it will be visible and crop the rest of the non pinned items

I don't think it's possible to offset regular items - it's done automatically by macOS.

**Comment by alexjousse** [comment]
Hey!
My two cents on your implementation:
* Pinned items at the top are a good idea for me because if we pin them, it means it's more important to the user than older items that have been pasted before.
* When opening Maccy, I would see first unpinned item selected not to modify the behavior of the app

**Comment by p0deje** [comment]
Added in 0.9.0 using ALT+p.

**Comment by mithunsridharan** [comment]
The pins are awesome. Could you also add a feature to keep the pins sorted on the front end based on the keys set in the preferences, please? The current sorting is based on the time of first or last copy or the number of copies, but not by the keys set when common clipboard items are pinned. That would immensely enhance this product for me. Thanks!

**Comment by chunyang-wen** [comment]
It seems pin is not working? When open Maccy using shortcut, press alt + p not working. Do not see them in the preference panel:
<img width="549" alt="image" src="https://user-images.githubusercontent.com/2911134/223348908-c09840b6-f2d0-47f9-8c0d-6aa55e72f795.png">

I just wonder how can I debug this? I am using 0.25.0 version of Maccy on Ventura.

**Comment by p0deje** [comment]
@chunyang-wen What keyboard layout do you use? It could be related to #485

**Comment by chunyang-wen** [comment]
Have read from the issue all the way to the end. I am using just the keyboard of MBP and my language is English.

OS: 13.2.1, M1Max
Maccy: 0.25.0,14

Option + p (no shift pressed) does not pin the items.

**Comment by chunyang-wen** [comment]
Found the root cause, the shortcut is registered by another app iShot. Thanks @p0deje I will update the other thread

**Comment by dhruvkhurmi** [comment]
THE SAME ISSUE PIN IS NOT WORKING. @p0deje @chunyang-wen @FernandoMiguel @mithunsridharan 

Can someone please help, I am a student and it would be very useful.

**Comment by namhap629** [comment]
I'm having the same problem of not being able to pin when pressing option + P, every time I press this character (π) appears. If only there was a shortcut setting button for this section, I could pin the information with another shortcut.
<img width="495" alt="Ảnh màn hình 2024-05-06 lúc 19 26 16" src="https://github.com/p0deje/Maccy/assets/169054453/32f08783-1682-4fde-850d-9580bf0bb5f0">

**Comment by p0deje** [comment]
@namhap629 Yes, check preferences to change the shortcut
